client/core/types.go

```diff
@@ -117,6 +117,7 @@ type User struct {
 	Initialized        bool                       `json:"inited"`
 	SeedGenerationTime uint64                     `json:"seedgentime"`
 	Assets             map[uint32]*SupportedAsset `json:"assets"`
+	FiatRates          map[uint32]float64         `json:"fiatRates"`
```

Comment:
If we can figure out a slightly different encoding here, something like

```Go
FiatRates    map[uint32]*float64
```

and then make a `nil` entry for disabled sources, then we can get rid of the `(*Core).FiatRateSources` method and the changes to `(*WebServer).handleSettings` (all page templates already have access to the `core.User`).

A similar approach might be to provide the rate as part of a struct, and similarly use `nil` to indicate disabled.

```Go
FiatRates map[uint32]*struct{
    Rate float64
}
```
Either way, it's a cleaner interface, I think.

Reply:
Uhm, currently the `map[string]bool` returned by `(*Core).FiatRateSources` method is useful for sending the rate source `name/token` to the frontend `settings page`, where js can manipulate it for use in enabling and disabling rate source.